go/pkg/filemetadata/filemetadata.go

```diff
@@ -30,6 +31,32 @@ type FileError struct {
 	Err        error
 }
 
+// External xattr package can be mocked for testing through this interface
+type xattributeAccessorInterface interface {
```

Comment:
Actually, do you need to mock this in the test? If our tests don't create files at all and rather mock the file creation process, we would also want to mock the fact that we set / get xattributes.

But our tests already have the ability to create temporary files on disk, so why not just directly set / get xattributes from the actual file on disk and skip the mock?

Reply:
See previous response (no guarantees can be made about the file system)
Technically I should also have unit tests for isSupported() returning false, and getXAttr() returning an error which can only be done consistently with mocking

Reply:
Right, sg!